src/plugins/intel_cpu/src/nodes/transpose.cpp

```diff
@@ -382,12 +381,7 @@ void Transpose::execute(dnnl::stream strm) {
         auto &dstMemPtr = getChildEdgeAt(0)->getMemoryPtr();
         auto &srcMemPtr = getParentEdgeAt(INPUT_DATA_IDX)->getMemoryPtr();
 
-        int MB = 0;
-        if (isDynamicNode()) {
-            MB = srcMemPtr->getStaticDims()[0];
-        } else {
-            MB = batchToProcess();
-        }
+        int MB = srcMemPtr->getStaticDims()[0];
 
         execPtr->exec(this, srcMemPtr, dstMemPtr, MB);
```

Comment:
it also would be nice to remove `batch` from executor API (here and in other places). However can be done as part of general executors refactoring activity.

Reply:
Yes, it does make sense to handle as a separate activity.